theforeman.org/pipelines/lib/packaging.groovy

```diff
@@ -481,17 +481,12 @@ def rsync_debian(user, ssh_key, suite, component, deb_paths) {
     }
 }
 
-def rsync_to_yum_stage(collection, version) {
-    def ssh_key = '/home/jenkins/workspace/staging_key/rsync_yumrepostage_key'
+def rsync_to_yum_stage() {
+    git url: "https://github.com/theforeman/theforeman-rel-eng", poll: false
 
-    if (!fileExists('upload_stage_rpms')) {
-        git url: "https://github.com/theforeman/theforeman-rel-eng", poll: false
-    }
+    sh "./generate_stage_repository"
 
-    sh """
-        export RSYNC_RSH="ssh -i ${ssh_key}"
-        export VERSION=${version}
-        export PROJECT=${collection}
-        ./upload_stage_rpms
-    """
+    sshagent(['yum-stage']) {
```

Comment:
This will need some manual migration, but allows us to drop this bit:
https://github.com/theforeman/foreman-infra/blob/801376af314f44108b4a851edc9e2d2173f7c247/puppet/modules/jenkins_node/manifests/packaging/rpm.pp#L44-L48